src/components/usertask-component/form-fields/date-form-field.tsx

```diff
@@ -18,19 +19,34 @@ export class DateFormField implements IFormField {
     return this.formField.id;
   }
 
+  // tslint:disable-next-line
+  private readonly _inputValidator: DateInputValidator = new DateInputValidator('/^\s*(3[01]|[12][0-9]|0?[1-9])\.(1[012]|0?[1-9])\.((?:19|20)\d{2})\s*$/');
+
   public componentWillLoad(): void {
     this.value = this.formField.defaultValue;
   }
 
   public render(): any {
-    return <div class='form-group'>
-      <label htmlFor={this.formField.id}>{this.formField.label}</label>
-      <input type='text' data-provide='datepicker' class='form-control'
-        id={this.formField.id} value={this.value} onFocus={(event: any): void => this._handleChange(event)} />
-    </div>;
+    return (
+      <div class= 'form-group'>
+        <label htmlFor={this.formField.id}>{this.formField.label}</label>
+        <input type='text' data-provide='datepicker' class='form-control' maxlength='10' placeholder='--.--.----'
+          id={this.formField.id} value={this.value} onFocus={(event: any): void => this._handleChange(event)}
+          onKeyDown={(event: any): void => this._handleKeyDown(event)}></input>
+
+      </div>
+    );
   }
 
   private _handleChange(event: any): void {
     this.value = event.target.value;
   }
+
+  private _handleKeyDown(event: any): void {
```

Comment:
```suggestion
  private _handleKeyDown(event: KeyEvent): void {
```

or something like that

Reply:
```
  private _handleKeyDown(event: KeyboardEvent): void {
    const value: string = this.value + event.key;
    const eventTarget = (event.target as HTMLInputElement);

    const isValidInput: boolean = (!this._inputValidator.validateKey(event.keyCode, eventTarget.value) && !this._inputValidator.isValid(value));

    if (isValidInput) {
      event.preventDefault();
    }
  }
```